crates/iroha_test_network/src/lib.rs

```diff
@@ -339,6 +339,15 @@ impl NetworkBuilder {
             commit_time = SumeragiParameters::default().commit_time();
         }
 
+        // Unoptimized profile requires additional resources.
+        if cfg!(debug_assertions) {
```

Comment:
Unoptimized profile of what? Of built WASMs? Of `irohad`?

This code is compiled as part of test code, and it has no relation to how WASMs and `irohad` were compiled. Therefore, setting this parameter by `debug_assertion` cfg is not a correct way.

I would suggest either:

- To set this parameter per-test, using `NetworkBuilder::with_genesis_instruction`
- If the profile in question is the profile of WASMs themselves, than store it somewhere after building the WASMs (e.g. `wasm_samples/target/.../profile.toml`), expose it alongside with `iroha_test_samples::load_sample_wasm` (e.g. `iroha_test_samples::sample_wasm_profile() -> iroha_wasm_builder::Profile`), and then use this knowledge in the tests/the test network itself.

Reply:
Good idea.